### PR TITLE
[plugin]support log4j2 AsyncLogger

### PR DESCRIPTION
**Pull request by xiaoy00**
Please answer these questions before submitting pull request

- Why submit this pull request?
- [1 ] Bug fix
- [ ] New feature provided
- [ ] Improve performance

- Related issues

___
### Bug fix
- Bug description.
there are 2 ways to implement Async log. but now this plugin just 1. so if i use Log4jContextSelector=org.apache.logging.log4j.core.async.AsyncLoggerContextSelector
to set async log . plugin doesn't work.

- How to fix?
intercept RingBufferLogEvent.setMessage(), i can get traceId,then its ok.
___
### New feature or improvement
- Describe the details and related test reports.

### CODE REVIEW

**Comment by wu-sheng** [comment]
@zhaoyuguang Could you check this locally?

File: apm-sniffer/apm-toolkit-activation/apm-toolkit-log4j-2.x-activation/src/main/java/org/apache/skywalking/apm/toolkit/activation/log/log4j/v2/x/async/RingBufferLogEventInstrumentation.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+                public boolean isOverrideArgs() {
+                    return false;
+                }
+            }
+        };
+    }
+
+    @Override
+    protected ClassMatch enhanceClass() {
+        return MultiClassNameMatch.byMultiClassMatch(ENHANCE_CLASS);
```
Comment (zhaoyuguang) [inline apm-sniffer/apm-toolkit-activation/apm-toolkit-log4j-2.x-activation/src/main/java/org/apache/skywalking/apm/toolkit/activation/log/log4j/v2/x/async/RingBufferLogEventInstrumentation.java]: Just use more clearly API:`NameMatch.byName` here,  can work?

File: apm-sniffer/apm-toolkit-activation/apm-toolkit-log4j-2.x-activation/src/main/java/org/apache/skywalking/apm/toolkit/activation/log/log4j/v2/x/async/RingBufferLogEventMethodInterceptor.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+public class RingBufferLogEventMethodInterceptor implements InstanceMethodsAroundInterceptor {
+
+
+    @Override
+    public void beforeMethod(EnhancedInstance objInst, Method method, Object[] allArguments, Class<?>[] argumentsTypes,
+                             MethodInterceptResult result) throws Throwable {
+        if (objInst instanceof EnhancedInstance) {
+            EnhancedInstance instances = objInst;
+            instances.setSkyWalkingDynamicField(ContextManager.getGlobalTraceId());
+        }
```
Comment (zhaoyuguang) [inline apm-sniffer/apm-toolkit-activation/apm-toolkit-log4j-2.x-activation/src/main/java/org/apache/skywalking/apm/toolkit/activation/log/log4j/v2/x/async/RingBufferLogEventMethodInterceptor.java]: Line `EnhancedInstance instances = objInst;` is not necessary.
And  you just want to `RingBufferLogEvent#private void setMessage(Message msg)` the msg is `msg instanceof ReusableMessage`?

Can we test that can work? 
```
@Override
public ElementMatcher<MethodDescription> getMethodsMatcher() {
    return named(ENHANCE_METHOD).and(takesArgumentWithType(0, ENHANCE_CLASS));
}
```

**Comment by enoch1024** [comment]
@wu-sheng @zhaoyuguang  When I use `Log4jContextSelector=org.apache.logging.log4j.core.async.AsyncLoggerContextSelector`
to set async log , plugin still doesn't work. I think it's similar to this #6815

**Comment by wu-sheng** [comment]
> @wu-sheng @zhaoyuguang When I use `Log4jContextSelector=org.apache.logging.log4j.core.async.AsyncLoggerContextSelector`
> to set async log , plugin still doesn't work. I think it's similar to this #6815

There is no chance async logger could work. Unless we hijack the `log#debug` by using the agent plugin mechanism.

**Pull request merged**